newsroom/notifications/models.py

```diff
@@ -0,0 +1,34 @@
+from datetime import datetime
+from dataclasses import dataclass
+from typing import Annotated, Optional
+from superdesk.core.resources import ResourceModel
+from superdesk.core.resources.fields import ObjectId, Field
+from superdesk.core.resources.validators import validate_data_relation_async
+from newsroom.users.module import users_resource_config
+
+
+user_validated_type = Annotated[ObjectId, validate_data_relation_async(users_resource_config.name)]
+
+
+class Notification(ResourceModel):
+    item: str | ObjectId
+    # TODO-async: replace the one above with the one below once `items` are migrated to async framework
+    # item: Annotated[Optional[ObjectId], validate_data_relation_async("items")] = None
+
+    user: user_validated_type
+    resource: Optional[str] = ""
+    action: Optional[str] = ""
+    data: Optional[dict] = {}
+
+
+@dataclass
+class Topic:
+    topic_id: ObjectId
```

Comment:
Should this validate the Topic with ID exists?

Reply:
I had it like that but I changed it because of some broken tests. Then I wondered the same. I can change the tests for sure but I was not sure if that should be the right behaviour. I'll adjust it.